scripts/irods/test/test_iadmin.py

```diff
@@ -2138,27 +2140,30 @@ def tearDown(self):
 
     # Issue 3862:  test that CAT_STATEMENT_TABLE_FULL is not encountered during parallel rebalance
     def test_rebalance__ticket_3862(self):
-
+        resource_name = "repl"
         def check_and_remove_rebalance_visibility_metadata(attempts):
             print("checking rebalance visibility metadata [{0} attempts remaining]".format(attempts))
-            self.assertTrue(attempts != 0, msg="rebalance visibility metadata not in catalog")
-            out,_,_ = self.admin.run_icommand(['imeta','ls','-R','repl'])
-            if "rebalance_operation" not in out:
-                check_and_remove_rebalance_visibility_metadata(attempts - 1)
-            else:
-                self.admin.assert_icommand("imeta rmw -R repl rebalance_operation % %") # b/c #3683
+            self.assertGreater(attempts, 0, msg="rebalance visibility metadata not in catalog")
+            resource_metadata_query = "select META_RESC_ATTR_NAME, META_RESC_ATTR_VALUE, META_RESC_ATTR_UNITS " \
+                                      f"where RESC_NAME = '{resource_name}'"
+            resource_avus = json.loads(self.admin.run_icommand(["iquery", resource_metadata_query])[0].strip())
```

Comment:
Just want to put on your radar that GenQuery2 limits the resultset to 256 rows when the query doesn't define a limit.

Reply:
Gotcha. If this ever has more than one AVU on it, then that's actually a problem. Perhaps I can insert an assertion to make sure of that here and we'll be good.

Reply:
Sure. Doesn't hurt.

Reply:
Done. Test passed